falcon-regression/merlin/src/test/java/org/apache/falcon/regression/nativeScheduler/NativeSchedulerAPITest.java

```diff
@@ -0,0 +1,306 @@
+        bundles[0].generateUniqueBundle(this);
+        bundles[0].setProcessWorkflow(aggregateWorkflowDir);
+        bundles[0].submitClusters(prism);
+        bundles[0].setProcessConcurrency(2);
+        bundles[0].setProcessValidity(startTime, endTime);
+        bundles[0].setProcessPeriodicity(1, Frequency.TimeUnit.minutes);
+
+        //Setting input and output feed as null (time based scheduling)
+        processMerlin = bundles[0].getProcessObject();
+        processMerlin.setInputs(null);
+        processMerlin.setOutputs(null);
+        LOGGER.info(processMerlin.toString());
+
+        // Submit process
+        ServiceResponse response = prism.getProcessHelper().submitEntity(processMerlin.toString());
+        AssertUtil.assertSucceeded(response);
+
+        // Schedule with prism
+        response = prism.getProcessHelper().schedule(processMerlin.toString(), null,
+                "properties=falcon.scheduler:native");
+        AssertUtil.assertSucceeded(response);
+
+        // Schedule with server
+        response = cluster1.getProcessHelper().schedule(processMerlin.toString(), null,
+                "properties=falcon.scheduler:native");
+        AssertUtil.assertSucceeded(response);
+
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void tearDown() {
+        removeTestClassEntities();
+    }
+
+    /*Suspend and resume entity and check that it is successful.
+     */
+    @Test
+    public void entitySuspendResumeTest() throws Exception {
+        //Suspend entity via prism
+        AssertUtil.assertSucceeded(prism.getProcessHelper().suspend(processMerlin.toString()));
+
+        Assert.assertTrue(Util.parseResponse(prism.getProcessHelper()
+                .getStatus(processMerlin.toString())).getMessage()
+                .contains("SUSPENDED"));
+
+        //Resume entity via prism
+        AssertUtil.assertSucceeded(prism.getProcessHelper().resume(processMerlin.toString()));
+
+        Assert.assertTrue(Util.parseResponse(prism.getProcessHelper()
+                .getStatus(processMerlin.toString())).getMessage()
+                .contains("RUNNING"));
+
+        //Suspend entity via server
+        AssertUtil.assertSucceeded(cluster1.getProcessHelper().suspend(processMerlin.toString()));
+
+        Assert.assertTrue(Util.parseResponse(cluster1.getProcessHelper()
+                .getStatus(processMerlin.toString())).getMessage()
+                .contains("SUSPENDED"));
+
+        //Resume entity via server
+        AssertUtil.assertSucceeded(cluster1.getProcessHelper().resume(processMerlin.toString()));
+
+        Assert.assertTrue(Util.parseResponse(cluster1.getProcessHelper()
+                .getStatus(processMerlin.toString())).getMessage()
+                .contains("RUNNING"));
+    }
+
+    /*Test for entity definition, list, dependency, update and summary API and check that it is successful.
+     */
+    @Test
+    public void entityDefinitionListDependencyUpdateSummaryTest() throws Exception {
```

Comment:
Ideally removeTestClassEntities() deletes all the entities and throws exception if delete fails. That is why didnt add the test for the same. I don't see the need of explicitly adding delete test case.